src/postgres_client/postgres_client_account_index.rs

```diff
@@ -0,0 +1,419 @@
+                error!("{}", msg);
+                return Err(AccountsDbPluginError::AccountsUpdateError { msg });
+            }
+
+            measure.stop();
+            inc_new_counter_debug!(
+                "accountsdb-plugin-postgres-update-owner-index-us",
+                measure.as_us() as usize,
+                10000,
+                10000
+            );
+            inc_new_counter_debug!(
+                "accountsdb-plugin-postgres-update-owner-index-count",
+                self.batch_size,
+                10000,
+                10000
+            );
+        }
+        Ok(())
+    }
+
+    /// Execute the token mint index bulk insert query.
+    pub fn bulk_insert_token_mint_index(&mut self) -> Result<(), AccountsDbPluginError> {
+        if self.pending_token_mint_index.len() == self.batch_size {
+            let mut measure =
+                Measure::start("accountsdb-plugin-postgres-prepare-mint-index-values");
+
+            let mut values: Vec<&(dyn types::ToSql + Sync)> =
+                Vec::with_capacity(self.batch_size * TOKEN_INDEX_COLUMN_COUNT);
+            for j in 0..self.batch_size {
+                let index = &self.pending_token_mint_index[j];
+                values.push(&index.owner);
+                values.push(&index.inner_key);
+                values.push(&index.slot);
+            }
+            measure.stop();
+            inc_new_counter_debug!(
+                "accountsdb-plugin-postgres-prepare-mint-index-values-us",
+                measure.as_us() as usize,
+                10000,
+                10000
+            );
+
+            let mut measure =
+                Measure::start("accountsdb-plugin-postgres-update-mint-index-account");
+            let client = self.client.get_mut().unwrap();
+            let result = client.client.query(
+                client.bulk_insert_token_mint_index_stmt.as_ref().unwrap(),
+                &values,
+            );
+
+            self.pending_token_mint_index.clear();
+
+            if let Err(err) = result {
+                let msg = format!(
+                    "Failed to persist the update of account to the PostgreSQL database. Error: {:?}",
+                    err
+                );
+                error!("{}", msg);
+                return Err(AccountsDbPluginError::AccountsUpdateError { msg });
+            }
+
+            measure.stop();
+            inc_new_counter_debug!(
+                "accountsdb-plugin-postgres-update-mint-index-us",
+                measure.as_us() as usize,
+                10000,
+                10000
+            );
+            inc_new_counter_debug!(
+                "accountsdb-plugin-postgres-update-mint-index-count",
+                self.batch_size,
+                10000,
+                10000
+            );
+        }
+        Ok(())
+    }
+
+    /// Generic function to queue the token owner index for bulk insert.
+    fn queue_token_owner_index_gen<G: GenericTokenAccount>(
```

Comment:
nit: not too sure what the `_gen` suffix stands for, seems like we can just remove it